test/Grpc.Net.Client.Tests/UserAgentGeneratorTests.cs

```diff
@@ -0,0 +1,134 @@
+                operatingSystem: "osx");
+            Assert.AreEqual($"grpc-dotnet/2.4.1 (.NET 5.0.7; CLR 5.0.7; netstandard2.0; osx x64)", userAgent);
+        }
+
+        [Test]
+        public void HandlesVersionWithGitHash()
+        {
+            var userAgent = UserAgentGenerator.GetUserAgentString(
+                processArch: Architecture.X64,
+                runtimeVersion: Version.Parse("5.0.7"),
+                assemblyVersion: "2.4.1-dev+5325faf",
+                runtimeInformation: ".NET 5.0.7",
+                frameworkName: ".NETStandard,Version=v2.0",
+                operatingSystem: "osx");
+            Assert.AreEqual($"grpc-dotnet/2.4.1-dev (.NET 5.0.7; CLR 5.0.7; netstandard2.0; osx x64)", userAgent);
+        }
+
+        [Test]
+        public void HandlesNoVersion()
+        {
+            var userAgent = UserAgentGenerator.GetUserAgentString(
+                processArch: Architecture.X64,
+                runtimeVersion: Version.Parse("5.0.7"),
+                assemblyVersion: string.Empty,
+                runtimeInformation: ".NET 5.0.7",
+                frameworkName: ".NETStandard,Version=v2.0",
+                operatingSystem: "osx");
+            Assert.AreEqual($"grpc-dotnet (.NET 5.0.7; CLR 5.0.7; netstandard2.0; osx x64)", userAgent);
+        }
+
+        [Test]
+        public void HandlesNoTargetFramework()
+        {
+            var userAgent = UserAgentGenerator.GetUserAgentString(
+                processArch: Architecture.X64,
+                runtimeVersion: Version.Parse("5.0.7"),
+                assemblyVersion: string.Empty,
+                runtimeInformation: ".NET 5.0.7",
+                frameworkName: string.Empty,
+                operatingSystem: "osx");
+            Assert.AreEqual($"grpc-dotnet (.NET 5.0.7; CLR 5.0.7; osx x64)", userAgent);
+        }
+
+        [Test]
+        public void HandlesNoRuntimeInfo()
+        {
+            var userAgent = UserAgentGenerator.GetUserAgentString(
+                processArch: Architecture.Arm64,
+                runtimeVersion: Version.Parse("5.0.7"),
+                assemblyVersion: string.Empty,
+                runtimeInformation: string.Empty,
+                frameworkName: string.Empty,
+                operatingSystem: "windows");
+            Assert.AreEqual($"grpc-dotnet (CLR 5.0.7; windows arm64)", userAgent);
+        }
+
+        [Test]
+        public void HandlesMonoRuntimeInfo()
+        {
+            var userAgent = UserAgentGenerator.GetUserAgentString(
+                processArch: Architecture.X64,
+                runtimeVersion: null,
+                assemblyVersion: string.Empty,
+                runtimeInformation: "Mono 6.12.0.140 (2020-02/51d876a041e Thu Apr 29 10:44:55 EDT 2021)",
+                frameworkName: string.Empty,
+                operatingSystem: "osx");
+            Assert.AreEqual($"grpc-dotnet (Mono 6.12.0.140; osx x64)", userAgent);
```

Comment:
`Environment.Version` isn't actually supported on Mono from my attempts but there's an alternative way to get the CoreCLR version if we want consistency here.

Reply:
in what sense it's not supported? I'm pretty sure I've been able to retrieve its value on mono and it has returned something that seemed reasonable.
Also, it must exist on mono and return something since Environment.Version existed in .NET Framework since version 1.1 and mono has to implement the API somehow.